cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java

```diff
@@ -285,20 +292,39 @@ public String cql2pgJson(String cql) throws QueryValidationException {
    * @throws QueryValidationException
    */
   public SqlSelect toSql(String cql) throws QueryValidationException {
+    return toSql(cql, 0, 0);
+  }
+
+  public SqlSelect toSql(String cql, int offset, int limit) throws QueryValidationException {
     try {
       CQLParser parser = new CQLParser();
       CQLNode node = parser.parse(cql);
-      return toSql(node);
+      return toSql(node, offset, limit);
     } catch (IOException|CQLParseException e) {
       throw new QueryValidationException(e);
     }
   }
 
-  private SqlSelect toSql(CQLNode node) throws QueryValidationException {
+  private SqlSelect toSql(CQLNode node, int offset, int limit) throws QueryValidationException {
+    String table = null;
+    String column = null;
+    int pos = jsonField.indexOf('.');
+    if (pos != -1) {
+      table = jsonField.substring(0, pos);
+      column = jsonField.substring(pos + 1);
+    }
     if (node instanceof CQLSortNode) {
-      return toSql((CQLSortNode) node);
+      return toSqlSort((CQLSortNode) node, offset, limit, table, column);
+    }
+    String where = pg(node);
+    StringBuilder sql = new StringBuilder(SELECT_STR + column + FROM_STR + table + WHERE_STR + where);
+    if (offset > 0) {
+      sql.append(OFFSET_STR + offset);
+    }
+    if (limit >= 0) {
+      sql.append(LIMIT_STR + limit);
```

Comment:
StringBuilder has memory and performance advantages, but only if you replace `+` String concatenation by append().
```suggestion
    StringBuilder sql = new StringBuilder().append(SELECT_STR).append(column)
                                           .append(FROM_STR).append(table)
                                           .append(WHERE_STR).append(where);
    if (offset > 0) {
      sql.append(OFFSET_STR).append(offset);
    }
    if (limit >= 0) {
      sql.append(LIMIT_STR).append(limit);
```